activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1160,8 +1162,7 @@ def check_if_method_has_arguments!(method_name, args)
 
     def structurally_compatible_for_or?(other)
       Relation::SINGLE_VALUE_METHODS.all? { |m| send("#{m}_value") == other.send("#{m}_value") } &&
-        (Relation::MULTI_VALUE_METHODS - [:extending]).all? { |m| send("#{m}_values") == other.send("#{m}_values") } &&
-        (Relation::CLAUSE_METHODS - [:having, :where]).all? { |m| send("#{m}_clause") != other.send("#{m}_clause") }
+        (Relation::MULTI_VALUE_METHODS - [:extending]).all? { |m| send("#{m}_values") == other.send("#{m}_values") }
```

Comment:
I don't remember exactly, but adding this line back makes tests fail:
```
ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:15:in `test_or_identity'

bin/test test/cases/relation/or_test.rb:13

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:10:in `test_or_with_relation'

bin/test test/cases/relation/or_test.rb:8

.E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/lib/active_record/querying.rb:10:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:39:in `test_or_without_left_where'

bin/test test/cases/relation/or_test.rb:37

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:81:in `test_or_on_loaded_relation'

bin/test test/cases/relation/or_test.rb:76

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:44:in `test_or_without_right_where'

bin/test test/cases/relation/or_test.rb:42

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:63:in `test_or_when_grouping'

bin/test test/cases/relation/or_test.rb:60

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/models/post.rb:47:in `block in <class:Post>'
    /Users/tamird/src/rails/activerecord/lib/active_record/scoping/named.rb:158:in `instance_exec'
    /Users/tamird/src/rails/activerecord/lib/active_record/scoping/named.rb:158:in `block (2 levels) in scope'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation.rb:343:in `scoping'
    /Users/tamird/src/rails/activerecord/lib/active_record/scoping/named.rb:158:in `block in scope'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/delegation.rb:88:in `public_send'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/delegation.rb:88:in `block in method_missing'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation.rb:343:in `scoping'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/delegation.rb:88:in `method_missing'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:73:in `test_or_inside_named_scope'

bin/test test/cases/relation/or_test.rb:71

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:68:in `test_or_with_named_scope'

bin/test test/cases/relation/or_test.rb:66

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:50:in `test_or_preserves_other_querying_methods'

bin/test test/cases/relation/or_test.rb:47

..E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:29:in `test_or_with_bind_params'

bin/test test/cases/relation/or_test.rb:28

E

ArgumentError: Relation passed to #or must be structurally compatible
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:647:in `or!'
    /Users/tamird/src/rails/activerecord/lib/active_record/relation/query_methods.rb:642:in `or'
    /Users/tamird/src/rails/activerecord/test/cases/relation/or_test.rb:25:in `test_or_with_null_right'
```

I think this used to work because empty `"#{m}_clause"`s would not compare equal, but do now since they are the same frozen object.